common/src/app/common/pages/changes.cljc

```diff
@@ -385,15 +385,27 @@
 
 (defmethod process-change :mov-objects
   [data {:keys [parent-id shapes index page-id component-id ignore-touched after-shape]}]
-  (letfn [(calculate-invalid-targets [objects shape-id]
+  (letfn [(nested-components? [objects shape-id]
+             (let [children          (cph/get-children-with-self objects shape-id)
+                   child-components  (into #{} (keep :component-id children))
+                   parents           (cph/get-parents-with-self objects parent-id)
+                   parent-components (into #{} (->> parents
+                                                    (filter :main-instance?)
+                                                    (map :component-id)))]
```

Comment:
here you also can replace

```
(into #{} (->> parents
               (filter :main-instance?)
               (map :component-id)))
```

```
(let [xform (comp (filter :main-instance?) 
                  xf-keep-component-id)]
  (into #{} xform parents))
```

Remember, that transducers is a form of reusable and composable way to define transformations independently of the underlying datastructure.

Reply:
transducers are my nemesis...

perhaps some day i will always remember to use them